packages/amplify-cli/src/lib/migrate-project.js

```diff
@@ -0,0 +1,188 @@
+  try {
+    let projectConfigFilePath;
+    try {
+      projectConfigFilePath = getProjectConfigFilePath();
+    } catch (e) {
+      // New project, hence not able to find the amplify dir
+      return;
+    }
+
+    const projectConfig = JSON.parse(fs.readFileSync(projectConfigFilePath));
+    if (projectConfig.projectPath && await prompt.confirm('We detected the project was initialized using an older version of the CLI. Do you want to migrate the project, so that it is compatible with the latest version of the CLI?')) {
+      // This is an older project & migration is needed
+      generateNewProjectConfig(projectConfig, projectConfigFilePath);
+      generateLocalEnvInfo(projectConfig);
+      generateAwsLocalInfo();
+      generateTeamProviderInfo();
+      generateBackendConfig();
+      generateGitIgnoreFile();
+    }
+
+    // Give each category a chance to migrate their respective files
+
+    const categoryMigrationTasks = [];
+    const amplifyMetafilePath = getAmplifyMetaFilePath();
+    const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+    const initializedCategories = Object.keys(amplifyMeta);
+
+    const categoryPlugins = {};
+    plugins.forEach((plugin) => {
+      if (plugin.name.includes('category')) {
+        const strs = plugin.name.split('-');
+        const categoryName = strs[strs.length - 1];
+        categoryPlugins[categoryName] = plugin.directory;
+      }
+    });
+
+    const availableCategory = Object.keys(plugins).filter(key =>
+      initializedCategories.includes(key));
+
+    availableCategory.forEach((category) => {
+      try {
+        const { migrateCategoryFiles } = require(categoryPlugins[category]);
+        if (migrateCategoryFiles) {
+          categoryMigrationTasks.push(() => migrateCategoryFiles());
+        }
+      } catch (e) {
+        print.warning(`Could not run migrateProject function for ${category} category`);
+      }
+    });
+
+    spinner.start('Migrating your project');
+    await Promise.all(categoryMigrationTasks);
+    spinner.succeed('Migrated your project successfully.');
+  } catch (e) {
+    spinner.fail('There was an error migrating your project.');
+    throw e;
+  }
+}
+
+function generateNewProjectConfig(projectConfig, projectConfigFilePath) {
+  const newProjectConfig = {};
+
+  Object.assign(newProjectConfig, projectConfig);
+  // These attributes are now stores in amplify/.config/local-env-info.json
+  delete newProjectConfig.projectPath;
+  delete newProjectConfig.defaultEditor;
+
+  // Modify frontend handler
+  const frontendPluginPath = Object.keys(projectConfig.frontendHandler)[0];
+  const frontendPlugin = frontendPluginPath.split('/')[frontendPluginPath.split('/').length - 1];
+  const frontend = frontendPlugin.split('-')[frontendPlugin.split('-').length - 1];
+
+  newProjectConfig.frontend = frontend;
+  delete newProjectConfig.frontendHandler;
+
+  // Modify provider handler
+  const provider = Object.keys(projectConfig.providers)[0];
```

Comment:
Is this wise?

Reply:
Yes, there's no other provider other cloudformation. Checked npm.

Reply:
For now 😉

Reply:
This breaks the cli pluggability.